packages/vite/src/node/plugins/html.ts

```diff
@@ -138,83 +131,134 @@ export const isAsyncScriptMap = new WeakMap<
 } {
-  let src: AttributeNode | undefined
+  let src: Token.Attribute | undefined
+  let sourceCodeLocation: Token.Location | undefined
   let isModule = false
   let isAsync = false
-  for (let i = 0; i < node.props.length; i++) {
-    const p = node.props[i]
-    if (p.type === NodeTypes.ATTRIBUTE) {
-      if (p.name === 'src') {
-        src = p
-      } else if (p.name === 'type' && p.value && p.value.content === 'module') {
-        isModule = true
-      } else if (p.name === 'async') {
-        isAsync = true
-      }
+  for (let i = 0; i < node.attrs.length; i++) {
+    const p = node.attrs[i]
+    if (p.name === 'src') {
+      src = p
+      sourceCodeLocation = node.sourceCodeLocation?.attrs!['src']
+    } else if (p.name === 'type' && p.value && p.value === 'module') {
+      isModule = true
+    } else if (p.name === 'async') {
+      isAsync = true
     }
   }
-  return { src, isModule, isAsync }
+  return { src, sourceCodeLocation, isModule, isAsync }
+}
+
+const attrValueWithQuotesStartRE = /=[\s\t\n\r]*"/
+
+export function overwriteAttrValue(
+  s: MagicString,
+  sourceCodeLocation: Token.Location,
+  newValue: string
+): MagicString {
+  const srcString = s.slice(
+    sourceCodeLocation!.startOffset,
+    sourceCodeLocation!.endOffset
+  )
+  const valueWithQuotes = srcString.match(attrValueWithQuotesStartRE)
+  const valueOffset =
+    1 + (valueWithQuotes ? srcString.indexOf('"') : srcString.indexOf('='))
+  s.overwrite(
+    sourceCodeLocation.startOffset + valueOffset,
+    sourceCodeLocation.endOffset + (valueWithQuotes ? -1 : 0),
+    newValue,
+    { contentOnly: true }
+  )
+  return s
 }
 
 /**
- * Format Vue @type {CompilerError} to @type {RollupError}
+ * Format parse5 @type {ParserError} to @type {RollupError}
  */
 function formatParseError(
-  compilerError: CompilerError,
+  parserError: ParserError,
   id: string,
   html: string
 ): RollupError {
-  const formattedError: RollupError = { ...(compilerError as any) }
-  if (compilerError.loc) {
-    formattedError.frame = generateCodeFrame(
-      html,
-      compilerError.loc.start.offset
-    )
-    formattedError.loc = {
-      file: id,
-      line: compilerError.loc.start.line,
-      column: compilerError.loc.start.column
-    }
+  const formattedError: RollupError = {
+    code: parserError.code,
+    message: `parse5 error code ${parserError.code}`
+  }
+  formattedError.frame = generateCodeFrame(html, parserError.startOffset)
+  formattedError.loc = {
+    file: id,
+    line: parserError.startLine,
+    column: parserError.startCol
   }
   return formattedError
 }
 
 function handleParseError(
-  compilerError: CompilerError,
+  parserError: ParserError,
   html: string,
   filePath: string
 ) {
+  switch (parserError.code) {
+    case 'missing-doctype':
+      // ignore missing DOCTYPE
+      return
+    case 'abandoned-head-element-child':
+      // Accept elements without closing tag in <head>
+      return
+    case 'duplicate-attribute':
```

Comment:
At least for this we could throw a warning 
But on the other side: linters can do this job
I mean we also use vue-tsc

Reply:
See other comments about this, I think we should tackle warnings (if we do so) in another PR